faucet/src/faucet.cpp

```diff
@@ -40,16 +40,20 @@ namespace faucets {
 
     check(faucet_itr != _faucet.end(), "Faucet does not exist");
 
-    if(current_time_point() - faucet_itr->started_at >= seconds(faucet_itr->interval.sec_since_epoch())) {
+    time_point now = current_time_point();
+    if(now > faucet_itr->until) {
+      uint32_t intervalSecs = faucet_itr->interval.sec_since_epoch();
+      float windowGap = (now.sec_since_epoch() - faucet_itr->until.sec_since_epoch()) / intervalSecs;
+      
+      time_point newUntil = faucet_itr->until + seconds(intervalSecs) + time_point(seconds(floor(windowGap) * intervalSecs));
```

Comment:
this line makes the correction window time if for some reason the faucet account doesn't claim its corresponding tokens